staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers_test.go

```diff
@@ -202,6 +202,67 @@ func TestSetCRDCondition(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "set new condition which doesn't have lastTransitionTime set",
+			crdCondition: []CustomResourceDefinitionCondition{
+				{
+					Type:               Established,
+					Status:             ConditionTrue,
+					Reason:             "Accepted",
+					Message:            "the initial names have been accepted",
+					LastTransitionTime: metav1.Date(2018, 1, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			newCondition: CustomResourceDefinitionCondition{
+				Type:    Established,
+				Status:  ConditionFalse,
+				Reason:  "NotAccepted",
+				Message: "Not accepted",
+			},
+			expectedcrdCondition: []CustomResourceDefinitionCondition{
+				{
+					Type:               Established,
+					Status:             ConditionFalse,
+					Reason:             "NotAccepted",
+					Message:            "Not accepted",
+					LastTransitionTime: metav1.Date(2018, 1, 2, 0, 0, 0, 0, time.UTC),
+				},
+			},
+		},
+		{
+			name: "append new condition which doesn't have lastTransitionTime set",
+			crdCondition: []CustomResourceDefinitionCondition{
+				{
+					Type:               Established,
+					Status:             ConditionTrue,
+					Reason:             "Accepted",
+					Message:            "the initial names have been accepted",
+					LastTransitionTime: metav1.Date(2018, 1, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			newCondition: CustomResourceDefinitionCondition{
+				Type:    Terminating,
+				Status:  ConditionFalse,
+				Reason:  "NeverEstablished",
+				Message: "resource was never established",
+			},
+			expectedcrdCondition: []CustomResourceDefinitionCondition{
+				{
+					Type:               Established,
+					Status:             ConditionTrue,
+					Reason:             "Accepted",
+					Message:            "the initial names have been accepted",
+					LastTransitionTime: metav1.Date(2018, 1, 1, 0, 0, 0, 0, time.UTC),
+				},
+				{
+					Type:               Terminating,
+					Status:             ConditionFalse,
+					Reason:             "NeverEstablished",
+					Message:            "resource was never established",
+					LastTransitionTime: metav1.Date(2018, 2, 1, 0, 0, 0, 0, time.UTC),
```

Comment:
I'm just curious how this works...shouldn't `metav1.NewTime(time.Now())` give the current local time? How is it `metav1.Date(2018, 2, 1, 0, 0, 0, 0, time.UTC)` everytime?

Reply:
Sorry, somehow missed this reply. :(